src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -325,9 +325,8 @@ public bool SkipPastAny(ReadOnlySpan<T> values)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public bool SkipPastAny(T value0, T value1, T value2, T value3)
+        public int SkipPastAny(T value0, T value1, T value2, T value3)
```

Comment:
Note the API signature change. @JeremyKuhne, I think we should return the number of elements skipped rather than a bool stating whether we skipped any or not. This change would apply to the other Skip APIs too. Let me know if you are fine with this.

This is required for the JsonReader.

Reply:
As long as it doesn't significantly impact the perf I have no problem with that.

Reply:
I would also say that we should be consistent across all of the APIs. They should always return count when they advance the reader.

Reply:
> They should always return count when they advance the reader.

> This change would apply to the other Skip APIs too.

I agree. Let me know if you are willing to make the changes to the BufferReader APIs.

Reply:
I am willing, but don't block on me. (I.e. you're welcome to make the change.)